src/Symfony/Component/DependencyInjection/Compiler/ResolveDefinitionTemplatesPass.php

```diff
@@ -96,11 +97,12 @@ private function resolveArguments(ContainerBuilder $container, array $arguments,
      */
     private function resolveDefinition(ContainerBuilder $container, DefinitionDecorator $definition)
     {
-        if (!$container->hasDefinition($parent = $definition->getParent())) {
+        try {
+            $parentDef = $container->findDefinition($parent = $definition->getParent());
```

Comment:
btw, this new feature must have tests covering it and a changelog entry (and it would be great to submit it as a dedicated PR)